### ISSUE TITLE
aws cloudformation package creates wrong URIs when working in china regions

### ISSUE THREAD

**Issue by rscheumann** [aws/aws-cli#3635]
### Context
I am deploying CloudFormation stacks into the Chinese regions; these are managed by AWS China, which support a subset of the services provided by AWS US, but is a separate entity and not AWS US.

### Action taken
The command

`aws cloudformation package --profile my-china-region-credentials --template master.yml --s3-bucket unique-bucket-name --output-template-file master-packaged.yml`

will package the master.yml template. It will rewrite TemplateURLs (to nested stacks) from local paths to an URI to an S3 bucket:

`TemplateURL: "./vpc.yml"`

becomes

`TemplateURL: https://s3.amazonaws.com/unique-bucket-name/0fa1483ed8df2859b23935c875861a2b.template`

### Expected result
As can be seen packaging does not take care of the China regions in Beijing and Ningxia. The URIs for S3 bucket in those regions start with `https://s3.cn-northwest-1.amazonaws.com.cn` instead of `https://s3.amazonaws.com`. 

If the packaged template gets deployed, CloudFormation will fail and issue the error message

`S3 error: The specified bucket does not exist`

which is expectable since the URIs are just wrong.

`aws cloudformaton package` should detect if a Chinese region is targeted and use a different URI pattern.

**Comment by dstufft** [comment]
@sanathkr thoughts?

**Comment by yngndrw-sage** [comment]
I'm getting a similar issue, but for non-Chinese regions.

CloudFormation is outputting a template with URLs in the format of:
`https://s3.amazonaws.com/<BucketName>/<Hash>.template`

... But this results in an error while creating the nested stack:
`S3 error: Unable to get the object https://s3.amazonaws.com/<BucketName>/<Hash>.template`

... And navigating to this results in an error:
PermanentRedirect
The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint.
`<BucketName>.s3.amazonaws.com`


It looks like the S3 URL format has changed and the CLI hasn't been updated to match?

**Comment by onelaview** [comment]
I just encountered this issue myself today. 😢 

In my case, I have a CloudFormation template to deploy several common IAM policies for an entire organization. This template will need to deploy to several AWS accounts, one of these is on AWS China.

The template keeps growing and it just hits template body size limit of 51,200 bytes. I decided to break down the stack to nested stacks and suddenly hit this bug causing me unable to deploy to AWS China.
